### PR TITLE
[MM-13979] Refactor PostInfo for E2E

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Refactor `PostInfo` component for E2E, adding `PostReaction` component.  

This is the first part of MM-13979.  Submitted on a separate PR for ease of review - https://github.com/mattermost/mattermost-webapp/pull/2355/files#r255449476

#### Ticket Link
Jira ticket: [MM-13979](https://mattermost.atlassian.net/browse/MM-13979)

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: components/post_view/post_info/post_info.jsx [hunk 1]
```diff
@@ -179,85 +152,41 @@ export default class PostInfo extends React.PureComponent {
...
-            comments = (
+        const showCommentIcon = fromAutoResponder ||
+        (!isSystemMessage && (isMobile || hover || (!post.root_id && Boolean(this.props.replyCount)) || this.props.isFirstReply));
+        const commentIconExtraClass = isMobile ? '' : 'pull-right';
+        let commentIcon;
+        if (showCommentIcon) {
+            commentIcon = (
                 <CommentIcon
-                    idPrefix='commentIcon'
+                    idPrefix={'commentIcon'}
```
Comment (jespino) [inline components/post_view/post_info/post_info.jsx]: not needed the `{}`

File: components/post_view/post_info/post_info.jsx [hunk 2]
```diff
@@ -297,11 +226,12 @@ export default class PostInfo extends React.PureComponent {
...
-        let flagIcon;
-        if (!isEphemeral && !post.failed && !isSystemMessage && (this.props.hover || this.props.isFlagged)) {
-            flagIcon = (
+        const showFlagIcon = !isEphemeral && !post.failed && !isSystemMessage && (this.props.hover || this.props.isFlagged);
+        let postFlagIcon;
+        if (showFlagIcon) {
+            postFlagIcon = (
                 <PostFlagIcon
-                    idPrefix='centerPostFlag'
+                    idPrefix={'centerPostFlag'}
```
Comment (jespino) [inline components/post_view/post_info/post_info.jsx]: same here

File: components/post_view/post_info/post_info.jsx (hunk 1 above)
Comment (saturninoabril) [reply components/post_view/post_info/post_info.jsx]: You're right, now updated :)

File: components/post_view/post_info/post_info.jsx [hunk 3]
```diff
@@ -345,13 +275,11 @@ export default class PostInfo extends React.PureComponent {
             );
         }
 
+        // timestamp should not be a permalink if the post has been deleted, is ephemeral message, or is pending
+        const isPermalink = !(isEphemeral || Posts.POST_DELETED === post.state || ReduxPostUtils.isPostPendingOrFailed(post));
```
Comment (sudheerDev) [inline components/post_view/post_info/post_info.jsx]: Nit: We can have this check inside the if clause as before.

**Comment by sudheerDev** [comment]
Minor nit as posted but i felt guilty to even point that out 😜.

File: components/post_view/post_info/post_info.jsx (hunk 3 above)
Comment (saturninoabril) [reply components/post_view/post_info/post_info.jsx]: Good catch, that's totally valid and I really appreciate it.  Thanks!

Now updated.

**Comment by saturninoabril** [comment]
@sudheerDev re: clean up, It just happened to be necessary for [E2E testability PR](https://github.com/mattermost/mattermost-webapp/pull/2355).  It's easier to change props of those affected components having it this way :smile:.

**Pull request merged**